test/functional/normal/put_spec.lua

```diff
@@ -15,7 +15,7 @@ local dedent = helpers.dedent
 local getreg = funcs.getreg
 
 local function reset()
-  clear()
+  execute('enew!')
```

Comment:
You're right, I missed that. For clarity, let's put it in a setup() inside the describe() block.

> should I put a teardown() call to ensure I leave the state clean for the next test group (double check to so that if one convention is forgotten stuff still holds).

Probably ok to leave it out, we expect all tests to set their expected state.